generics/services/project.js

```diff
@@ -322,10 +322,97 @@ const pullSolutionsFromProgramComponents = function (solutionId,tenantId) {
   });
 };
 
+
+/**
+ * @function createChildProjectTemplate
+ * @description Creates a child project template by calling the Project Service API.
+ *              Sends the project template external IDs, user details, and program info.
+ *
+ * @param {Array<string>} projectTemplateExternalIds - List of project template external IDs.
+ * @param {Object} userDetails - Information about the user making the request.
+ * @param {string} programId - External ID of the program.
+ * @param {boolean} isExternalProgram - Whether the program is external (true/false).
+ *
+ * @returns {Promise<Object>} Resolves with the API response if successful,
+ *                            or with `{ success: false }` in case of errors or timeouts.
+ */
+const createChildProjectTemplate = function (projectTemplateExternalIds,userDetails,programId,isExternalProgram) {
+  return new Promise(async (resolve, reject) => {
+    try {
+
+      // isExternalProgram - We have to negate this boolean before sending to project service . 
+      //                     if isExternalProgram true for samiksha service, for project service it is false
+
+      // Ensure isExternalProgram is always false if passed as true
+      let IsExternalProgramFalse = (isExternalProgram === true) ? false : isExternalProgram;
```

Comment:
write the logic like if true send false if false then true. basically you have to switch